### ISSUE TITLE
As a dev I want to be able to create and commit a branch since sometimes I start coding and later think that this belongs in its own branch

### ISSUE THREAD

**Issue by togakangaroo** [togakangaroo/ezgit#5]
maybe
```
ezgit branch-commit "cool new feature"
```
which would be the same as
```
git checkout -b cool-new-feature
git add -A
git commit -m "start cool new feature"
```

It's just such a common thing would be nice to have it in a single command. 

Thought bubble: might be a good idea to provide a mechanism for this to also move the most recent N commits to the new branch?

**Comment by daviraimundo** [comment]
So you've been making changes and you didn't spin off a new branch before you did so but now you want to. It makes sense to me. But, you can't always checkout branches depending on the changes you've made. It may be safer to stash first, checkout the new branch and then pop from stash.

The 2nd part I'm not sure what you mean. All the commits of the current branch are automatically brought to a new branch when creating.

**Comment by togakangaroo** [comment]
Yeah, however it works under the hood is fine.

For the second part I'm thinking of the situation when I already shave a commit or two (unpushed) and think "oh crap, I'm on default aren't I? I should have created a branch".

It's useful when I think something is a bugfix but turns into a whole refactoring, or when I just plain screw up

**Comment by daviraimundo** [comment]
so would it would reset the current branch's HEAD to the specified commit after creating the new one? If so, that makes sense to me.

**Comment by togakangaroo** [comment]
Yeah, I think so. I think it will also commit anything in your working directory as needed (maybe it will have to ask you to confirm or something)
